tensorflow/go/operation.go

```diff
@@ -123,6 +148,67 @@ func (p Output) c() C.TF_Output {
 
 func (p Output) canBeAnInput() {}
 
+// NumConsumers returns the number of inputs that consume this output.
+func (p Output) NumConsumers() int {
+	return int(C.TF_OperationOutputNumConsumers(p.c()))
+}
+
+// Consumers returns the inputs that consume this output.
+func (p Output) Consumers() []OpInput {
+	max := p.NumConsumers()
+	consumers := make([]C.TF_Input, max)
+	n := C.TF_OperationOutputConsumers(p.c(), (*C.TF_Input)(unsafe.Pointer(&consumers[0])), C.int(max))
+	consumers = consumers[:int(n)]
+
+	var inputs []OpInput
+	for _, consumer := range consumers {
+		inputs = append(inputs, OpInput{
+			Index: int(consumer.index),
+			Op: &Operation{
+				c: consumer.oper,
+				g: p.Op.g,
+			},
+		})
+	}
+
+	return inputs
+}
+
+// OpInput is one of the inputs to an operation.
+type OpInput struct {
+	// Op is the Operation that consumes this Input.
+	Op *Operation
+
+	// Index specifies the index of the input within the Operation.
+	Index int
+}
+
+func (p OpInput) c() C.TF_Input {
+	if p.Op == nil {
+		// Attempt to provide a more useful panic message than "nil
+		// pointer dereference".
+		panic("nil-Operation. If the Output was created with a Scope object, see Scope.Err() for details.")
```

Comment:
This panic message doesn't seem helpful :). Perhaps something like:

```go
panic("nil-Operation. This Consumer object was not created by a call to Output.Consumers")
```

or something like that?

Reply:
Updated